kernel/src/main/java/org/kframework/kompile/DefinitionParsing.java

```diff
@@ -369,12 +378,61 @@ private Module resolveNonConfigBubbles(Module module, Scanner scanner, RuleGramm
         }
     }
 
+    /**
+     * Replace all the targeted Bubbles from the definition if they can be found in caches.
+     * @param def    The Definition with Bubbles.
+     * @param isRule true if it should target non config Bubbles, false if it should parse only config bubbles
+     * @return A new Definition object with Bubbles replaced by the appropriate Sentence type.
+     */
+    private Definition resolveCachedBubbles(Definition def, boolean isRule) {
+        RuleGrammarGenerator gen = new RuleGrammarGenerator(def);
+        return DefinitionTransformer.from(m -> {
+            if (stream(m.localSentences()).noneMatch(s -> s instanceof Bubble && (isRule || ((Bubble) s).sentenceType().equals(configuration))))
+                return m;
+            ParseCache cache = isRule ? loadCache(gen.getRuleGrammar(m)) : loadCache(gen.getConfigGrammar(m));
```

Comment:
Here's a third place we're calling loadCache

Reply:
There are now only two places, once when looking for cached bubbles, and another when doing parsing.
I've done some profiling and it seems that the most expensive step is actually `ParseInModule.getExtensionModule()`
`loadCache` didn't even register. I could avoid calling it twice by combining `resolveBubbles` and `resolveCachedBubbles` but it doesn't seem to be a problem, so at the moment I prefer to have them separate for clarity.